torch/distributed/_tensor/_sharding_prop.py

```diff
@@ -37,6 +38,14 @@ def _length(obj) -> int:
     return len(obj)
 
 
+class LocalLRUCache(threading.local):
+    def __init__(self, user_function: Callable) -> None:
+        self.cache = lru_cache(None)(user_function)
+
+    def apply(self, *args, **kwargs) -> object:
```

Comment:
can we instead overwrite the `__call__` method, so that we can save this `.apply` and still use `self.propagate_op_sharding(op_info.schema)`?

Reply:
yeah that's better.